wes_client/client.py

```diff
@@ -0,0 +1,121 @@
+from logging import Logger
+import socket
+
+import requests
+from requests import Session
+import urllib3
+
+class wesClient:
+    def __init__(self,url:str, base_path:str, token: Optional[str] = None,)-> None:
+        """Class Constructor"""
+        self.url = url
+        self.base_path = base_path
+        self.headers = {"Accept": "application/json"}
+        self.session = Session()
+        self.session.headers.update(self.headers)  
+        self.token = token
+        
+    def get_runs(self,token:Optional[str] = None):
+        url = f"{self.url}{self.base_path}/runs"
+        response = self.session.get(url)
+        if token:
+            self.token = token
```

Comment:
You are changing the value of an instrance attribute from a method, thus the method has a side effect here. I know TRS-cli etc. are full of these, but it's really best to avoid these. If you MUST set an instance attribute, use a dedicated setter method.